### DISCUSSION TITLE
Il documento non è una fattura elettronica

### DISCUSSION THREAD

**Discussion by donlegro** [fattureincloud/api#100] [Q&A] [answered]
Salve,
inviando questo tracciato la chiamata issued_documents non ha problemi
Mentre la chiamata xml_verify mi restituisce l'errore in oggetto.
Qual è il dato mancante o sbagliato? Io credo di averli messi tutti corretti.
Vorrei avere conferma sul campo [ei_type] => TD01 perchè mi sembra strano da mettere nel gruppo [entity]
Grazie in anticipo.

`Array
(
    [id] => 0
    [type] => invoice
    [subject] => FATTURA IMMEDIATA 616B del 22/09/2022
    [visible_subject] => FATTURA IMMEDIATA 616B del 22/09/2022
    [entity] => Array
        (
            [id] => 0
            [type] => company
            [name] => VITTO ... 
            [address_street] => VIA ...
            [address_postal_code] => 70..
            [address_city] => ...
            [address_province] => BA
            [address_extra] => 
            [country] => Italia
            [email] => 
            [certified_email] => 
            [phone] => 
            [ei_code] => 6JXPS2J
            [vat_number] => IT05...
            [tax_code] => VTT...
            [e_invoice] => 1
            [ei_type] => TD01
        )

    [number] => 616
    [numeration] => B
    [date] => 2022-09-22
    [year] => 2022
    [currency] => Array
        (
            [id] => EUR
            [exchange_rate] => 1.00000
            [symbol] => Â€
            [html_symbol] => €
        )

    [language] => Array
        (
            [code] => it
            [name] => Italiano
        )

    [use_gross_prices] => 
    [rivalsa] => 0
    [cassa] => 0
    [withholding_tax] => 0
    [withholding_tax_taxable] => 0
    [other_withholding_tax] => 0
    [stamp_duty] => 0
    [rc_center] => 
    [notes] => 
    [use_split_payment] => 
    [e-invoice] => 1    
    [ei_data] => Array
        (
            [vat_kind] => I
            [invoice_number] => B/616
            [invoice_date] => 2022-09-22
            [payment_method] => MP01
            [bank_name] => BANCA
            [bank_iban] => 
            [bank_beneficiary] => 
        )

    [items_list] => Array
        (
            [0] => Array
                (
                    [id] => 44428
                    [product_id] => 3112
                    [code] => PRE-001
                    [name] => Piantine di PREZZEMOLO G. ITALIA CAPUT
                    [category] => 
                    [net_price] => 0.022
                    [discount] => 0
                    [qty] => 90000
                    [measure] => NR
                    [vat] => Array
                        (
                            [id] => 3
                            [value] => 10
                            [description] => 10%
                            [e_invoice] => 1
                            [ei_type] => 
                            [ei_description] => 
                        )

                    [not_taxable] => 
                    [apply_withholding_taxes] => 
                    [discount_highlight] => 0
                    [in_dn] => 
                    [stock] => 
                    [ei_raw] => Array
                        (
                            [AltriDatiGestionali] => Array
                                (
                                    [0] => Array
                                        (
                                            [TipoDato] => LOTTO
                                            [RiferimentoTesto] => CN1980B (NTR)
                                            [RiferimentoNumero] => 90000
                                        )

                                )

                        )

                )

        )

    [payments_list] => Array
        (
            [0] => Array
                (
                    [id] => 0
                    [due_date] => 2022-10-31
                    [amount] => 2178.00
                    [status] => not_paid
                    [paid_date] => 
                    [payment_account] => Array
                        (
                            [id] => 0
                        )

                )

        )

)
`

**Comment by donlegro** [comment]
Rispondo da solo perchè sbagliavo a scrivere e_invoce = true mettendo il trattino (e-invoice) invece dell'underscore (e_invoce).
Però rimane la seconda parte della domanda perchè vorrei avere conferma e capire l'utilità dei righi
[e_invoice] => true
[ei_type] => TD01
all'interno di entity
Grazie.

**Comment by emajo** [comment, marked as answer]
Ciao @donlegro `e_invoice` nel cliente specifica che il cliente è una pubblica amministrazione, mentre per quanto riguarda ei_type non dovrebbe essere salvato a livello cliente.

Resto a disposizione per ulteriori chiarimenti,
Emanuele

**Comment by donlegro** [comment]
Chiedo scusa ma non trovo riscontro nelle guide con quanto affermato in questa risposta e quindi chiedo ancora cortese conferma. 
Se ho capito bene:

`entity {
    e_invoice = true
}`
va messo solo se stiamo fatturando a un ente pubblico?


Grazie per le delucidazioni.

**Reply by emajo** [reply]
Esatto, stiamo preparando una guida per clienti/fornitori, aggiungeremo sicuramente questa informazione.
